bundle/permissions/permission_diagnostics_test.go

```diff
@@ -0,0 +1,159 @@
+	b := mockBundle([]resources.Permission{
+		{Level: "CAN_MANAGE", GroupName: "testgroup"},
+		{Level: "CAN_MANAGE", UserName: "alice@databricks.com"},
+	})
+
+	diags := ReportPermissionDenied(context.Background(), b, "testpath")
+	require.ErrorContains(t, diags.Error(), "include: alice@databricks.com")
+}
+
+func TestReportTerraformError1(t *testing.T) {
+	ctx := context.Background()
+	b := mockBundle([]resources.Permission{
+		{Level: "CAN_MANAGE", UserName: "alice@databricks.com"},
+	})
+	err := TryExtendTerraformPermissionError(ctx, b, errors.New(`Error: terraform apply: exit status 1
+
+Error: cannot update permissions: ...
+
+	with databricks_pipeline.my_project_pipeline,
+	on bundle.tf.json line 39, in resource.databricks_pipeline.my_project_pipeline:
+	39:       }`)).Error()
+	require.ErrorContains(t, err, string(diag.ResourcePermissionDenied))
+}
+
+func TestReportTerraformError2(t *testing.T) {
+	ctx := context.Background()
+	b := mockBundle([]resources.Permission{
+		{Level: "CAN_MANAGE", UserName: "alice@databricks.com"},
+	})
+	err := TryExtendTerraformPermissionError(ctx, b, errors.New(`Error: terraform apply: exit status 1
+
+Error: cannot read pipeline: User xyz does not have View permissions on pipeline 4521dbb6-42aa-418c-b94d-b5f4859a3454.
+
+	with databricks_pipeline.my_project_pipeline,
+	on bundle.tf.json line 39, in resource.databricks_pipeline.my_project_pipeline:
+	39:       }`)).Error()
+	require.ErrorContains(t, err, string(diag.ResourcePermissionDenied))
+}
+
+func TestReportTerraformError3(t *testing.T) {
+	ctx := context.Background()
+	b := mockBundle([]resources.Permission{
+		{Level: "CAN_MANAGE", UserName: "alice@databricks.com"},
+	})
+	err := TryExtendTerraformPermissionError(ctx, b, errors.New(`Error: terraform apply: exit status 1
+
+	Error: cannot read permissions: 1706906c-c0a2-4c25-9f57-3a7aa3cb8b90 does not have Owner permissions on Job with ID: ElasticJobId(28263044278868). Please contact the owner or an administrator for access.
+
+	with databricks_pipeline.my_project_pipeline,
+	on bundle.tf.json line 39, in resource.databricks_pipeline.my_project_pipeline:
+	39:       }`)).Error()
+	require.ErrorContains(t, err, string(diag.ResourcePermissionDenied))
+}
+
+func TestReportTerraformErrorNotOwner(t *testing.T) {
+	ctx := context.Background()
+	b := mockBundle([]resources.Permission{
+		{Level: "CAN_MANAGE", UserName: "alice@databricks.com"},
+	})
+	b.Config.RunAs = &jobs.JobRunAs{
+		UserName: "testuser@databricks.com",
+	}
+	err := TryExtendTerraformPermissionError(ctx, b, errors.New(`Error: terraform apply: exit status 1
+
+Error: cannot read pipeline: User xyz does not have View permissions on pipeline 4521dbb6-42aa-418c-b94d-b5f4859a3454.
+
+	with databricks_pipeline.my_project_pipeline,
+	on bundle.tf.json line 39, in resource.databricks_pipeline.my_project_pipeline:
+	39:       }`)).Error()
+	require.ErrorContains(t, err, string(diag.ResourcePermissionDenied))
+}
```

Comment:
The TF error parsing seems like it belongs in a separate file.

Reply:
Moved to terraform_errors.go.